pkg/storage/image.go

```diff
@@ -599,19 +600,30 @@ func GetImageService(ctx context.Context, store storage.Store, defaultTransport
 		seenRegistries[r] = true
 	}
 
+	systemRegistries, err := sysregistriesv2.FindUnqualifiedSearchRegistries(sc)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
(Non-blocking: This can be moved into the `else` branch of `if len(cleanRegistries) != 0`;

and the cleaning of `registries` to `cleanRegistries` can be moved into the `if` branch of the same.)